Jenkinsfile

```diff
@@ -28,26 +31,45 @@ pipeline {
                     }
                 }
                 stages {
+                    stage('Generate docker compose file') {
+                        agent {
+                            label 'linux'
+                        }
+                        when {
+                            not {
+                                environment name: 'IMAGE_TYPE', value: 'linux'
+                            }
+                        }
+                        steps {
+                            script {
+                                def imageTypeParts = env.IMAGE_TYPE.split('-')
+                                def windowsFlavor = "[\"${imageTypeParts[0]}\"]"
+                                def windowsVersion = "[\"${imageTypeParts[1]}\"]"
+                                def currentVersion = env.TAG_NAME ? env.TAG_NAME : '1.0.0'
+
+                                sh "WINDOWS_FLAVORS_TO_BUILD='$windowsFlavor' WINDOWS_VERSIONS_TO_BUILD='$windowsVersion' VERSION=$currentVersion ON_TAG=true make generate-compose-$IMAGE_TYPE"
```

Comment:
It is considered a good Jenkins pipeline practise to avoid direct groovy interpolation like this. You should wrap the `sh` step within a `withEnv` with a literal string passed as argument